federation/pkg/federation-controller/util/federated_informer.go

```diff
@@ -97,8 +100,17 @@ type FederatedInformer interface {
 // framework.DeletionHandlingMetaNamespaceKeyFunc as a keying function.
 type TargetInformerFactory func(*federation_api.Cluster, federation_release_1_4.Interface) (cache.Store, framework.ControllerInterface)
 
+type ClusterLifecycleHandler func(*federation_api.Cluster, FederationView, FederatedReadOnlyStore)
```

Comment:
If you don't see a need to any namespace and RS, then we should probably remove it.